apps/heft/src/plugins/TypescriptPlugin/TypescriptBuilder.ts

```diff
@@ -0,0 +1,1029 @@
+                    newLinkPath: path.join(outFolderPath, dtsMapFilename)
+                  });
+                }
+              }
+
+              if (!sourceFile.isDeclarationFile) {
+                const jsFilename: string = `${relativeFilenameWithoutExtension}.js`;
+                queueLinkOrCopy({
+                  linkTargetPath: path.join(cacheOutFolderPath, jsFilename),
+                  newLinkPath: path.join(outFolderPath, jsFilename)
+                });
+              }
+            }
+          }
+        }
+
+        await Promise.all(linkPromises);
+
+        return { linkCount };
+      }
+    );
+
+    this._terminal.writeVerboseLine(
+      `${shouldHardlink ? 'Hardlink' : 'Copy from cache'}: ${hardlinkDuration}ms (${hardlinkCount} files)`
+    );
+    //#endregion
+
+    let compilationFailed: boolean = false;
+    if (diagnostics.length > 0) {
+      this._terminal.writeErrorLine(
+        `Encountered ${diagnostics.length} TypeScript error${diagnostics.length > 1 ? 's' : ''}:`
+      );
+      for (const diagnostic of diagnostics) {
+        this._printDiagnosticMessage(ts, diagnostic, true);
+      }
+
+      compilationFailed = true;
+    }
+
+    if (tslintResult?.failures?.length) {
+      this._terminal.writeWarningLine(
+        `Encountered ${tslintResult!.failures.length} TSLint error${
+          tslintResult!.failures.length > 1 ? 's' : ''
+        }:`
+      );
+      for (const tslintFailure of tslintResult!.failures) {
+        const buildFolderRelativeFilename: string = path.relative(
+          this._configuration.buildFolder,
+          tslintFailure.getFileName()
+        );
+        const { line, character } = tslintFailure.getStartPosition().getLineAndCharacter();
+        const severity: string = tslintFailure.getRuleSeverity().toUpperCase();
+        this._terminal.writeWarningLine(
+          '  ',
+          Colors.yellow(`${severity}: ${buildFolderRelativeFilename}(${line + 1},${character + 1})`),
+          Colors.white(' - '),
+          Colors.yellow(`(${tslintFailure.getRuleName()}) ${tslintFailure.getFailure()}`)
+        );
+      }
+    }
+
+    if (compilationFailed) {
+      throw new Error('TypeScript compilation failed.');
+    }
+  }
+
+  private _printDiagnosticMessage(
+    ts: ExtendedTypescript,
+    diagnostic: Typescript.Diagnostic,
+    withIndent: boolean = false
+  ): void {
+    let terminalMessage: (string | IColorableSequence)[];
+    // Code taken from reference example
+    if (diagnostic.file) {
+      const { line, character } = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start!);
+      const message: string = ts.flattenDiagnosticMessageText(diagnostic.messageText, '\n');
+      const buildFolderRelativeFilename: string = path.relative(
+        this._configuration.buildFolder,
+        diagnostic.file.fileName
+      );
+      terminalMessage = [
+        Colors.red(`ERROR: ${buildFolderRelativeFilename}(${line + 1},${character + 1})`),
```

Comment:
Do we still prefer the parenthesized notation?
```
src/folder/File.ts(12,34)
```
`tsc` now defaults to "Unix" notation, which I think is supported by a wider range of editors:
```
src/folder/File.ts:12:34
```

This [ESLint page](https://eslint.org/docs/user-guide/formatters/) lists all the popular formats.  (The one we're using is called "visualstudio".)

Reply:
I don't have a strong opinion. I can change it.